src/range-slider/index.ts

```diff
@@ -0,0 +1,372 @@
+			onfocus: prepareCallback((prop, e1, e2) => this._genericCallback(prop, e1, e2), this.properties.onFocus),
+			onchange: prepareCallback((prop, e1, e2) => this._genericChangeCallback(prop, e1, e2), this.properties.onChange),
+			oninput: prepareCallback((prop, e1, e2) => this._genericChangeCallback(prop, e1, e2), this.properties.onInput),
+			onkeydown: this._onKeyDown,
+			onkeypress: this._onKeyPress,
+			onkeyup: this._onKeyUp,
+			onmousedown: this._onMouseDown,
+			onmouseup: this._onMouseUp,
+			ontouchstart: this._onTouchStart,
+			ontouchend: this._onTouchEnd,
+			ontouchcancel: this._onTouchCancel,
+			classes: [this.theme(css.input), fixedCss.nativeInput]
+		};
+	}
+
+	protected renderOutput(minValue: number, maxValue: number, percentValue: number[]): DNode {
+		const {
+			output,
+			outputIsTooltip = false
+		} = this.properties;
+
+		const outputNode = output ? output(minValue, maxValue) : `${minValue}, ${maxValue}`;
+
+		// output styles
+		let outputStyles: { left?: string; top?: string } = {};
+		if (outputIsTooltip) {
+			outputStyles = { left: `${Math.round((percentValue[0] + (percentValue[1] - percentValue[0]) / 2) * 100)}%` };
+		}
+
+		return v('output', {
+			classes: this.theme([css.output, outputIsTooltip ? css.outputTooltip : null]),
+			for: this._widgetId,
+			styles: outputStyles,
+			tabIndex: -1 /* needed so Edge doesn't select the element while tabbing through */
+		}, [outputNode]);
+	}
+
+	render(): DNode {
+		const {
+			disabled,
+			widgetId = this._widgetId,
+			invalid,
+			label,
+			labelAfter,
+			labelHidden,
+			max = 100,
+			min = 0,
+			readOnly,
+			required,
+			theme,
+			showOutput = false
+		} = this.properties;
+		const focus = this.meta(Focus).get('root');
+		let { minValue = min, maxValue = max } = this.properties;
+
+		minValue = Math.max(minValue, min);
+		maxValue = Math.min(maxValue, max);
+
+		const slider1Percent = (minValue - min) / (max - min);
+		const slider2Percent = (maxValue - min) / (max - min);
+
+		const slider1Size = slider1Percent + (slider2Percent - slider1Percent) / 2;
+		const slider2Size = 1 - slider1Size;
+
+		const size = this.meta(Dimensions).get('root');
+
+		const slider1Focus = this.meta(Focus).get('slider1');
+		const slider2Focus = this.meta(Focus).get('slider2');
+
+		const slider1 = v('input', {
+			...this._getInputProperties(true),
+			key: 'slider1',
+			value: `${minValue}`,
+			styles: {
+				clip: `rect(auto, ${Math.round((slider1Size) * size.client.width)}px, auto, auto)`
+			}
+		});
+		const slider2 = v('input', {
+			...this._getInputProperties(false),
+			key: 'slider2',
+			value: `${maxValue}`,
+			styles: {
+				clip: `rect(auto, auto, auto, ${Math.round((1 - slider2Size) * size.client.width)}px)`
+			}
+		});
+
+		const children = [
+			label ? w(Label, {
+				key: 'label',
+				theme,
+				disabled,
+				focused: focus.containsFocus,
+				invalid,
+				readOnly,
+				required,
+				hidden: labelHidden,
+				forId: widgetId
```

Comment:
Since this is pointing to the root `div`, and that div doesn't have any region/etc attributes to make it a landmark (which makes sense, since it isn't a landmark 😁), the `forId` isn't really doing anything. If you use `aria-labelledby` on the `input`s, you could drop this entirely